openspace.cfg

```diff
@@ -167,6 +167,19 @@ ModuleConfigurations = {
     },
     Space = {
         ShowExceptions = false
+    },
+    Exoplanets = {
+        ShowComparisonCircle = false,
+        ShowHabitableZone = true,
+        UseOptimisticZone = true,
+        -- OBS! All these paths must be set
+        -- @TODO (emmbr 2021-02-10) this comment can be removed once we have
+        -- documentation for modules
+        DataFolder = "${SYNC}/http/exoplanets_data/2/",
```

Comment:
I would suggest putting the setting of these values to be a lua script that is run in the onInitalize of the exoplanets_data.asset and exoplanets_textures.asset files. Right now these feel so hard coded to the asset being sync'd and never changing it's folder names, version numbers etc.

This would be similar to how the webgui.asset sets its cef route after it has synced the download of the html.

Reply:
Good point! I totally agree that setting them from the asset would be better. 

However, since the paths are not set to optional, not setting them in the `openspace.cfg` results in an uncaught SpecificationError (see issue #1502). Could get around this by setting the paths to optional, but the exoplanets stuff cannot be used without them, so there might be a risk of problems if a user tries to change things

 Also, if a user sets a custom path in the cfg file (to for example another texture), it will later be overwritten by the asset's `onInitialize`? 🤔 Not super clear to the user what's going on here, I think

Reply:
Having them not optional just to satisfy the SpecificationError seems not the correct way to go. Is the code even checking that these files exists, or just that the specification is met. It feels like "oh the rules say this so I'm doing this"

The nice thing about them being optional is that then we wouldn't have to even put them in our .cfg file. 

The asset could check the module properties with getPropertyValue to see if they are set before it sets them itself. 

I believe is the first time we are trying to consider replacing  sync/ asset via a variable in the module configurations which is why we are coming to this unsolved issue.

Reply:
The specification is only checking that the path is set, but I agree that it would be better to actually check that the files actually exist as well. Maybe a specialized `StringVerifier` that does this could be added for files

And yes, seems like the first time. But it seems to me like the nicer option, for now, is to make the properties optional and do as you suggested: check if there is a value with `getPropertyValue` before setting the paths. Maybe it's also worth discussing at our next meeting, to see how we want to handle these scenarios in the future

Reply:
Alex also pointed out that making the properties optional might actually make sense. "if I run OpenSpace without the exoplanets module or I am certain I never want to create any exoplanets it is an optional value since I can run happily without it"